src/components/useResource.tsx

```diff
@@ -0,0 +1,47 @@
+import { TResource } from '../types';
+import { useState, useEffect } from 'react';
+import firebase from '../firebase';
+
+/**
+ * Hook which provides a Resource record from Firebase
+ *
+ * @example
+ * const CharityDisplay = props: {dataPath: string} => {
+ *   const charityData: TResource | null = useResource(props.dataPath);
+ *   return charityData ? (
+ *     <div>
+ *       <h1>{charityData.charityName}</h1>
+ *       <p>{charityData.description}</p>
+ *     </div>
+ *   ) : (
+ *     <Spinner />
+ *   );
+ * };
+ */
+function useResource(dataPath: string): TResource | null {
```

Comment:
Not a big deal, but is there a reason to use `function` here instead of an arrow function? Wasn't sure if this was convention I haven't seen yet.

Reply:
hahaha what? writing a function?